changes/4767.housekeeping

```diff
@@ -0,0 +1 @@
+Update Job and App Developer docs to use the `nautobot.apps.jobs` module over the `nautobot.extras.jobs` module.
```

Comment:
```suggestion
Updated Job and App Developer docs to import from the `nautobot.apps` module throughout.
```

Once #4777 is merged this can change to a `.documentation` entry but we can handle that.